### PR TITLE
Minor cleanups in tests

### PR DESCRIPTION
**Pull request by hubertp**
[ci no changelog needed]

### Pull Request Description

Removing some leftovers from big PRs.

### CODE REVIEW

File: engine/runtime/src/test/scala/org/enso/compiler/test/semantic/TypeSignaturesTest.scala [hunk 1]
```diff
@@ -45,11 +45,7 @@ trait TypeMatchers {
       expr: IR.Expression
     ): Option[(Sig, IR.Expression, String)] = (sig, expr) match {
       case (Name(n), t: IR.Name.Literal) if n == t.name =>
-        if (n == t.name) {
```
Comment (kustosz) [inline engine/runtime/src/test/scala/org/enso/compiler/test/semantic/TypeSignaturesTest.scala:48]: This is real logic, making sure we verify names correctly

Comment (hubertp) [reply engine/runtime/src/test/scala/org/enso/compiler/test/semantic/TypeSignaturesTest.scala:48]: Yes, but it is already guarded by the condition one line above

Comment (hubertp) [reply engine/runtime/src/test/scala/org/enso/compiler/test/semantic/TypeSignaturesTest.scala:48]: Bringing back custom error, as discussed

File: engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala [hunk 2]
```diff
@@ -728,7 +728,9 @@ class Compiler(
           diagnostics.map(_._2.collect { case e: IR.Error => e }.length).sum
         val warnCount =
           diagnostics.map(_._2.collect { case e: IR.Warning => e }.length).sum
-        println(s"Aborting due to ${count} errors and ${warnCount} warnings.")
+        context.getOut.println(
```
Comment (hubertp) [inline engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala]: @kustosz I actually wonder if we should print this message at all? Seems spurious?

**Review by 4e6 — APPROVED** [review]

**Review by JaroslavTulach — APPROVED** [review]

**Pull request merged**